torch/csrc/jit/passes/graph_fuser.cpp

```diff
@@ -90,11 +90,11 @@ struct GraphFuser {
   GraphFuser(std::shared_ptr<Graph>& graph)
   : graph(graph) {}
 
-  bool isCuda(Node * node) {
+  int getDevice(Node * node) {
     if(node->kind() == kFusionGroup) {
-      return node->i(kis_cuda);
+      return node->i(kdevice);
     }
-    return node->output()->type()->expect<TensorType>()->device() != -1;
+    return node->output()->type()->expect<TensorType>()->device();
```

Comment:
Can you add a check that all inputs/outputs of a fusion group have a matching device?

Reply:
It is part of the `shouldFuse` check, `consumer_device == getDevice(producer->node())`

Reply:
Oh sorry, that was unclear. I meant as part of a lint pass